packages/remark-grid-tables/src/index.js

```diff
@@ -376,6 +376,47 @@ function extractTableContent (lines, linesInfos, hasHeader) {
   return table
 }
 
+function processCellLines (lines) {
+  const trimmedLines = []
+  let inCodeBlock = false
+  let leadingWhitespace = 0
+
+  for (let i = 0; i < lines.length; i++) {
+    let line = lines[i]
+
+    // Trim the end of the line
+    line = line.replace(/\s+$/, '')
+
+    // Check if we're entering or exiting a code block
+    if (line.trim().startsWith('```')) {
+      inCodeBlock = !inCodeBlock
+
+      // If we're entering a code block, remember the amount of leading whitespace
+      if (inCodeBlock) {
+        // Set how much whitespace to remoev from lines
+        // inside the code-block
+        leadingWhitespace = line.match(/^\s*/)[0].length
+      }
+
+      // Remove leading whitespace from the opening/closing code-block
+      // statement as well
+      line = line.substring(leadingWhitespace)
+    } else if (inCodeBlock) {
+      // If we're *already* in a code block, trim the start of the line
+      // by the amount of leading whitespace
+      line = line.substring(leadingWhitespace)
```

Comment:
Let's say I have the following Markdown:

```markdown
+------+----------------------+
| cell | `code                |
|      | with line break`     |
+------+----------------------+
|  ```text                    |
| line 1                      |
| line2                       |
|```                          |
+------+----------------------+
```

The output will have the first character of the code stripped for each line. I am quite unsure of what could be done in that case, maybe counting the number of whitespaces every time and if it is lesser than the number currently saved, update it?

Reply:
Yeah, you're right. But I don't see this as hard to address...
It "just" requires to use a regexp that removes "that many" space chars from the untrimmed string, if there are less... It should probably do nothing (let the user face their own doing).

If there are more, then it's all good..

Reply:
What's hard is to find which behavior is to be implemented, not to choose one ;)

That's also why I needed some time to review your changes: I checked how it is done in [`micromark-extension-gridtables`](https://github.com/adobe/micromark-extension-gridtables), which will likely be the plugin we will use when switching to the new micromark system. The way they implemented it (which actually delegates to `micromark` core), gives the same results as your approach, so let's do it like this!